### ISSUE TITLE
typing of `dataclasses.replace` and `copy.replace`

### ISSUE THREAD

**Issue by tpgillam** [astral-sh/ty#2938]
### Question

Consider the following example:

```python
import copy
import dataclasses


@dataclasses.dataclass
class X:
    a: int


x = X(3)
dataclasses.replace(x, a="moo")
copy.replace(x, a="moo")
x.__replace__(a="moo")
```

Currently, for ty and for pyright, only the last of these is flagged as an error:

```console
$ uv run pyright moo.py
/home/tom/burgas/moo.py
  /home/tom/burgas/moo.py:13:17 - error: Argument of type "Literal['moo']" cannot be assigned to parameter "a" of type "int" in function "__replace__"
    "Literal['moo']" is not assignable to "int" (reportArgumentType)
1 error, 0 warnings, 0 informations

$ uvx ty check moo.py
Installed 1 package in 7ms
error[invalid-argument-type]: Argument is incorrect
  --> moo.py:13:15
   |
11 | dataclasses.replace(x, a="moo")
12 | copy.replace(x, a="moo")
13 | x.__replace__(a="moo")
   |               ^^^^^^^ Expected `int`, found `Literal["moo"]`
   |
info: rule `invalid-argument-type` is enabled by default

Found 1 diagnostic
```

Might it be in scope for ty to synthesise the same type signatures for `dataclasses.replace` and/or `copy.replace`?

We like the static typing guarantees, although using `__replace__` directly seems stylistically suboptimal.

### Version

0.0.19

**Comment by tmke8** [comment]
I think the only thing missing here is for `copy.replace()` to use `ParamSpec` to use the synthesized signature of `__replace__`. There has been a PR open for this on typeshed since October: https://github.com/python/typeshed/pull/14819

**Comment by carljm** [comment]
Thanks for the report! Yes, I think this is entirely about the typeshed annotations for `dataclasses.replace` and `copy.replace`, I don't think there's any new feature needed in ty itself.

Will close this since I don't think it's actionable for ty, but I do think it's a good idea!
